source/OrderedTypes/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,875 @@
+  pr₂ (WEM-lemma P) (_ , y-has-lub-cond) not-P =
+    x-not-equal-y (≤-is-antisymmetric A (x-below-y)
+                                      (y-has-lub-cond (x , x-is-ub)))
+   where
+    x-is-ub : (x is-an-upper-bound-of (δ-fam x y P)) holds
+    x-is-ub (inl ✯) = ≤-is-reflexive A x
+    x-is-ub (inr in-P) = 𝟘-induction (not-P in-P)
+    
+\end{code}
+
+We now show that the two element poset is δ complete only if WEM holds.
+
+\begin{code}
+
+2-is-non-trivial : non-trivial-posets.is-non-trivial-poset 2-Poset
+2-is-non-trivial = (₀ , ₁ , ⋆ , zero-is-not-one)
+
+2-is-δ-complete-gives-WEM : {𝓥 : Universe}
+                          → δ-complete-poset.is-δ-complete {𝓤₀} {𝓤₀} 𝓥 2-Poset
+                          → WEM 𝓥
+2-is-δ-complete-gives-WEM {𝓥} i P P-is-prop = wem
+ where
+  open Joins (rel-syntax 2-Poset)
+  open δ-complete-poset 𝓥 2-Poset
+  open non-trivial-posets 2-Poset  
+
+  sup-from-δ-completeness : Σ s ꞉ ∣ 2-Poset ∣ₚ ,
+                          (s is-lub-of (δ-fam ₀ ₁ (P , P-is-prop))) holds
+  sup-from-δ-completeness = i ₀ ₁ ⋆ (P , P-is-prop)
+
+  sup-gives-wem : Σ s ꞉ ∣ 2-Poset ∣ₚ ,
+                           (s is-lub-of (δ-fam ₀ ₁ (P , P-is-prop))) holds
+                         → ¬ P + ¬ (¬ P)
+  sup-gives-wem (₀ , sup) =
+    inl (pr₁ (WEM-lemma 𝓥 2-is-non-trivial (P , P-is-prop)) sup)
+  sup-gives-wem (₁ , sup) =
+    inr (pr₂ (WEM-lemma 𝓥 2-is-non-trivial (P , P-is-prop)) sup)
+
+  wem : ¬ P + ¬ (¬ P)
+  wem = sup-gives-wem sup-from-δ-completeness
+
+\end{code}
+
+Since non-trivial is a negated concept it only has enough strength to derive
+WEM, we now introduce the stronger concept of positivity.
+
+\begin{code}
+
+module Positive-Posets (𝓤  𝓦  𝓥 : Universe) (A : Poset 𝓤 𝓦) where
+ 
+ open δ-complete-poset 𝓥 A
+ open Universal fe
+ open PosetReasoning A
+ open Joins (_≤_)
+
+ module positive-posets (i : is-δ-complete) where
+
+  _<_ : (x y : ∣ A ∣ₚ) → 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
+  x < y = (x ≤ y) holds
+        × ((z : ∣ A ∣ₚ)
+          → (y ≤ z) holds
+          → (P : Ω 𝓥)
+          → (z is-lub-of (δ-fam x z P)) holds
+          → P holds)
+
+  order-from-strictly-below : {x y : ∣ A ∣ₚ} → x < y → (x ≤ y) holds
+  order-from-strictly-below c = pr₁ c
+
+  sup-condition-from-strictly-below : {x y : ∣ A ∣ₚ}
+                                    → x < y
+                                    → ((z : ∣ A ∣ₚ)
+                                     → (y ≤ z) holds
+                                     → (P : Ω 𝓥)
+                                     → (z is-lub-of (δ-fam x z P)) holds
+                                     → P holds)
+  sup-condition-from-strictly-below x-strictly-below-y =
+     pr₂ x-strictly-below-y
+
+  strictly-below-implies-non-trivial : (x y : ∣ A ∣ₚ)
+                                     → is-δ-complete
+                                     → (x < y)
+                                     → (x ≤ y) holds × (x ≠ y)
+  pr₁ (strictly-below-implies-non-trivial x y i x-strictly-below-y) =
```

Comment:
Could you, instead of using copatterns, here and in other places, do the following instead please?

```
strictly-below-implies-non-trivial x y i x-strictly-below-y = I , II
 where
  I : ?
  I = ?
  II : ?
  II = ?
```
So that `I` and `II` make up the pair in the conclusion of the statement. Thanks!